privaterelay/middleware.py

```diff
@@ -123,3 +124,21 @@ def __call__(self, request):
         if first_visit is None and not request.user.is_anonymous:
             response.set_cookie("first_visit", datetime.now(timezone.utc))
         return response
+
+
+class RelayStaticFilesMiddleware(WhiteNoiseMiddleware):
+    def immutable_file_test(self, path, url):
+        """
+        Determine whether given URL represents an immutable file (i.e. a
+        file with a hash of its contents as part of its name) which can
+        therefore be cached forever.
+
+        All files outputed by next.js are hashed and immutable
+        """
+        if not url.startswith(self.static_prefix):
+            return False
+        name = url[len(self.static_prefix) :]
```

Comment:
This code to remove the static prefix is from the Whitenoise code:

https://github.com/evansd/whitenoise/blob/ae01e9a63dccebf20de48df03f8f8b9cb457f7d8/src/whitenoise/middleware.py#L179-L181

Since our `STATIC_ROOT` is `/`, this trims off the initial slash from the URL. This will still work if we use a different `STATIC_ROOT` in the future.

https://github.com/evansd/whitenoise/blob/ae01e9a63dccebf20de48df03f8f8b9cb457f7d8/src/whitenoise/middleware.py#L173-L181